.gitlab-ci.yml

```
@@ -367,3 +373,36 @@ publish:s3:mender-monitor:automatic:
   rules:
     - if: '$PUBLISH_MENDER_DIST_PACKAGES_AUTOMATIC == "true"'
   extends: .template:publish:s3:mender-monitor
+
+.template:publish:s3:mender-gateway:
+  stage: publish
+  image: debian:buster
+  dependencies:
+    # It depends on either of them
+    - build:packages:local-image
+    # - build:packages:upstream-image
+  before_script:
+    - apt update && apt install -yyq awscli
+    - *publish_helper_functions
+  script:
+    - echo "Publishing mender-gateway version ${MENDER_GATEWAY_VERSION} to s3://${S3_BUCKET_NAME_PRIVATE}/${S3_BUCKET_SUBPATH_GATEWAY}/${MENDER_GATEWAY_VERSION}/"
+    - aws s3 cp output/commercial/mender-gateway*.deb s3://${S3_BUCKET_NAME_PRIVATE}/${S3_BUCKET_SUBPATH_GATEWAY}/${MENDER_GATEWAY_VERSION}/
+    # Make copies in known destinations to be consumed by get.mender.io script:
+    # * For "experimental" channel, we make a copy of the newest master package in the master directory
+    # * For "stable" channel, we make a copy in a separate "latest" subdirectory of the latest tagged version
+    # This needs to be reworked before Mender 3.2, see MEN-5029 for more details.
```

I suggest we update this comment (and the original one) to something like:
```suggestion
    # This can to be reworked for better maintainability, see MEN-5029 for more details.
```